tests/test-utils/org.wso2.identity.apps.test.container/src/main/java/org/wso2/identity/apps/test/container/CypressTestContainer.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.Writer;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Cypress Integration Test Utils.
+ */
+public class CypressTestContainer {
+
+    private static final CypressTestContainer INSTANCE = new CypressTestContainer();
+
+    private static final Log LOG = LogFactory.getLog(CypressTestContainer.class);
+
+    private CypressTestContainer() {
+
+    }
+
+    /**
+     * Get the instance.
+     *
+     * @return instance.
+     */
+    public static CypressTestContainer getInstance() {
+
+        return INSTANCE;
+    }
+
+    /**
+     * Add or overwrite the environment configurations of cypress test suite.
+     *
+     * @param filePath File path of the cypress.env.json file.
+     * @param propertyName Name of the cypress environment property.
+     * @param propertyValue Value of the cypress environment property.
+     */
+    public void addOrOverwriteTestConfigProperty(Path filePath, String propertyName, String propertyValue) {
+
+        if (StringUtils.isBlank(propertyName)) {
+            throw new RuntimeException("Invalid config element propertyName.");
+        }
+
+        Gson gson = new Gson();
+
+        try (Reader reader = Files.newBufferedReader(filePath, StandardCharsets.UTF_8)) {
+            JsonObject envConfigJSON = gson.fromJson(reader, JsonObject.class);
+            if (envConfigJSON.get(propertyName) != null) {
+                envConfigJSON.remove(propertyName);
+            }
+            envConfigJSON.addProperty(propertyName, propertyValue);
+
+            try (Writer writer = Files.newBufferedWriter(filePath, StandardCharsets.UTF_8)) {
+                gson.toJson(envConfigJSON, writer);
+            }
+            LOG.info("Updated " + CypressTestConstants.CYPRESS_ENV_CONFIG_FILE + " file with key: " + propertyName +
+                    ", value: " + propertyValue);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to update " + CypressTestConstants.CYPRESS_ENV_CONFIG_FILE +
+                    " with key: " + propertyName + ", value: " + propertyValue);
+        }
+    }
+
+    /**
+     * Execute the identity apps cypress test suite.
+     *
+     * @param scriptPath Path of the script to run the test suite.
+     * @throws IOException When character encoding is not supported.
+     */
+    public void runTestSuite(Path scriptPath) throws IOException  {
+
+        if (!Files.exists(scriptPath)) {
+            throw new RuntimeException("Script `" + scriptPath.toAbsolutePath().toString() + "` does not exists!");
+        }
+
+        Process process;
+        if (System.getProperty("os.name").startsWith("Windows")) {
+            // TODO run bat file.
```

Comment:
We have a bat file created right? Why are we skipping for windows. Shall we fix the TODO ?

Reply:
There was no way for me to test the batch script and that's why I thought it would be better to skip it for now.